### PR TITLE
Better docs for empty/zeros/ones/full

### PR DESCRIPTION
**Pull request by jakevdp**
Part of #21461

### CODE REVIEW

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -3341,11 +3341,40 @@ def copy(a: ArrayLike, order: str | None = None) -> Array:
...
+
+  Args:
+    a: Array-like object with ``shape`` and ``dtype`` attributes.
+    shape: optionally override the shape of the created array.
+    dtype: optionally override the dtype of the created array.
+    device: (optional) :class:`~jax.Device` or :class:`~jax.sharding.Sharding`
+      to which the created array will be committed.
+
+  Returns:
+    array of the specified shape and dtype, on the specified device if specified.
```
Comment (mattjj) [inline jax/_src/numpy/lax_numpy.py:3360]: Is it intentional to sometimes write "Array" (like in "a: Array-like ...") and other times "array" (like here)? Maybe we should always capitalize?
```suggestion
    Array of the specified shape and dtype, on the specified device if specified.
```

File: jax/_src/numpy/lax_numpy.py [hunk 2]
```diff
@@ -3354,11 +3383,40 @@ def zeros_like(a: ArrayLike | DuckTypedArray,
...
```
Comment (mattjj) [inline jax/_src/numpy/lax_numpy.py:3402]:
```suggestion
    Array of the specified shape and dtype, on the specified device if specified.
```

File: jax/_src/numpy/lax_numpy.py [hunk 3]
```diff
@@ -3367,13 +3425,42 @@ def ones_like(a: ArrayLike | DuckTypedArray,
...
```
Comment (mattjj) [inline jax/_src/numpy/lax_numpy.py:3446]:
```suggestion
    Array of the specified shape and dtype, on the specified device if specified.
```

File: jax/_src/numpy/lax_numpy.py [hunk 4]
```diff
@@ -3387,10 +3474,43 @@ def _normalize_to_sharding(device: xc.Device | Sharding | None) -> Sharding | No
...
+  Args:
+    shape: int or sequence of ints specifying the shape of the created array.
+    fill_value: scalar or array with which to fill the created array.
+    dtype: optional dtype for the created array; defaults to the dtype of the
+      fill value.
+    device: (optional) :class:`~jax.Device` or :class:`~jax.sharding.Sharding`
+      to which the created array will be committed.
+
+  Returns:
+    array of the specified shape and dtype, on the specified device if specified.
```
Comment (mattjj) [inline jax/_src/numpy/lax_numpy.py:3493]:
```suggestion
    Array of the specified shape and dtype, on the specified device if specified.
```

File: jax/_src/numpy/lax_numpy.py [hunk 5]
```diff
@@ -3402,11 +3522,46 @@ def full(shape: Any, fill_value: ArrayLike,
...
+  Args:
+    a: Array-like object with ``shape`` and ``dtype`` attributes.
+    fill_value: scalar or array with which to fill the created array.
+    shape: optionally override the shape of the created array.
+    dtype: optionally override the dtype of the created array.
+    device: (optional) :class:`~jax.Device` or :class:`~jax.sharding.Sharding`
+      to which the created array will be committed.
+
+  Returns:
+    array of the specified shape and dtype, on the specified device if specified.
```
Comment (mattjj) [inline jax/_src/numpy/lax_numpy.py:3542]:
```suggestion
    Array of the specified shape and dtype, on the specified device if specified.
```

File: jax/_src/numpy/lax_numpy.py [hunk 6]
```diff
@@ -3423,31 +3578,108 @@ def full_like(a: ArrayLike | DuckTypedArray,
...
+  JAX implementation of :func:`numpy.ones`.
+
+  Args:
+    shape: int or sequence of ints specifying the shape of the created array.
+    dtype: optional dtype for the created array; defaults to floating point.
+    device: (optional) :class:`~jax.Device` or :class:`~jax.sharding.Sharding`
+      to which the created array will be committed.
+
+  Returns:
+    array of the specified shape and dtype, on the specified device if specified.
```
Comment (mattjj) [inline jax/_src/numpy/lax_numpy.py:3630]:
```suggestion
    Array of the specified shape and dtype, on the specified device if specified.
```

File: jax/_src/numpy/lax_numpy.py [hunk 7]
```diff
@@ -3423,31 +3578,108 @@ def full_like(a: ArrayLike | DuckTypedArray,
...
+  full of zeros.
+
+  Args:
+    shape: int or sequence of ints specifying the shape of the created array.
+    dtype: optional dtype for the created array; defaults to floating point.
+    device: (optional) :class:`~jax.Device` or :class:`~jax.sharding.Sharding`
+      to which the created array will be committed.
+
+  Returns:
+    array of the specified shape and dtype, on the specified device if specified.
```
Comment (mattjj) [inline jax/_src/numpy/lax_numpy.py:3668]:
```suggestion
    Array of the specified shape and dtype, on the specified device if specified.
```

**Review by mattjj — APPROVED** [review]

File: jax/_src/numpy/lax_numpy.py (hunk 1 above)
Comment (jakevdp) [reply jax/_src/numpy/lax_numpy.py]: Done, thanks!

**Pull request merged**